src/ray/util/event.h

```diff
@@ -137,9 +137,15 @@ class RayEventContext final {
                        const std::unordered_map<std::string, std::string> &custom_fields =
                            std::unordered_map<std::string, std::string>());
 
-  void SetCustomField(const std::string &key, const std::string &value);
+  // Only for test, isn't thread-safe with SetEventContext.
+  void ResetEventContext();
 
-  void SetCustomFields(const std::unordered_map<std::string, std::string> &custom_fields);
+  // If the key already exists, replace the value. Otherwise, insert a new item.
+  void AddCustomField(const std::string &key, const std::string &value);
+
+  // Add the `custom_fields` into the existing items.
+  // If the key already exists, replace the value. Otherwise, insert a new item.
+  void AddCustomFields(const std::unordered_map<std::string, std::string> &custom_fields);
```

Comment:
According to the semantics, I think "UpdateCustomFields" would be a better name. This is similar to Python dict update.